src/network/postgres_protocol_handler.cpp

```diff
@@ -730,7 +730,30 @@ size_t PostgresProtocolHandler::ReadParamValue(
         PL_ASSERT(param_values[param_idx].GetTypeId() != type::TypeId::INVALID);
       } else {
         // BINARY mode
-        switch (static_cast<PostgresValueType>(param_types[param_idx])) {
+        PostgresValueType pg_value_type = static_cast<PostgresValueType>(param_types[param_idx]);
+        switch (pg_value_type) {
+          case PostgresValueType::TINYINT: {
+            int int_val = 0;
+            for (size_t i = 0; i < sizeof(int); ++i) {
+              int_val = (int_val << 8) | param[i];
```

Comment:
(I followed some of the code to lower layers, but didn't get the clarity desired). However, if we need to pull data off the wire, are there not functions to do this, rather than this approach?